### DISCUSSION TITLE
Populate results in MissingSchemaError

### DISCUSSION THREAD

**Discussion by kekami** [strapi/strapi#6627] [Legacy Topics]
**Describe the bug**
I have a model called page and I am trying to use populate on one of the properties to get the actual objects rather than the ids. However, when I add anything other than an empty array as the second argument of findOne, I get `error MissingSchemaError: Schema hasn't been registered for model "page".
Use mongoose.model(name, schema)` 

```
// api/page/services/page.js
module.exports = {
  findOne(params, populate) {
    return strapi.query('page').model.findOne(params, ['foo.bar']);
  },
};
```
**Expected behavior**
I expect to receive

```
{
  // page stuff
  foo: {
   bar: {
     // bar stuff
   }
  }
}
```

rather than

```
{
  // page stuff
  foo: {
   bar: "5ee1ea57e6d7a41689b6ac42"
  }
}
```

**System**

- Node.js version: v12.16.1
- NPM version: 6.13.4
- Strapi version: 3.0.1
- Database: MongoDB shell version v4.2.6
- Operating system: MacOS 10.15.5

**Comment by derrickmehaffy** [comment]
You also need to add `foo` to that array. Moving this to a discussion as it's not a bug.

**Reply by kekami** [reply]
Thanks for the quick reply, but I get the `MissingSchemaError` if I put anything in the array, same for `['foo', 'foo.bar'], ['baz']` etc It only works if the array is empty.

**Reply by enzzoperez** [reply]
Hello, can you fix it?, I have the same error

**Reply by gtamas** [reply]
@derrickmehaffy 

How to use populate with relations  that produce arrays?

like:

```
// some record
{
contents: [
// related items from contents model / collectien
{...}
]
}
```

How do I define populate in this case?

`strapi.query('something', ['contents', 'contents.language']) `

produces a missing schema error. The "language" is a property of all related items from content and is a relation itself (points to a language record).

**Reply by ms-eztut** [reply]
@lauriejim
Same problem here. In our project we need to return only specific fields and relations depending on the user role.

**Comment by derrickmehaffy** [comment]
The population array only works on relation fields, not standard fields.

In the case of standard fields, we are looking into adding a feature (likely sometime in v4) to allow you to specifically request only certain fields. In the short term you will need to use this guide: https://strapi.io/documentation/v3.x/guides/custom-data-response.html
